src/main/java/radiounit/NokiaWcdmaRadioUnitReceiver.java

```diff
@@ -82,24 +84,26 @@ public void modifyCarrierNokiaWcdma(Integer carrierId, FrequencyBand frequencyBa
 		System.out.println(
 				"[NokiaWcdmaRadioUnitReceiver] modifyCarrierNokiaWcdma: " + carrierId + ", " + frequencyBand.getBand());
 
-		Carrier existingCarrier = carriers.get(carrierId);
-		if (existingCarrier == null) {
+		if (!this.carriers.containsKey(carrierId)) {
 			System.err.println("Carrier with ID [" + carrierId + "] does not exist!");
 			return;
 		}
 
-		removeCarrierInternal(carrierId);
-		existingCarrier.setFrequencyBand(frequencyBand);
-		addCarrierInternal(existingCarrier);
+		Carrier existingCarrier = this.carriers.get(carrierId);
+		
+		Objects.requireNonNull(existingCarrier).setFrequencyBand(frequencyBand);
```

Comment:
Small note. Did we want to add a String parameter to the requireNonNull methods to indicate what the issue would be?

e.g. "No carrier is linked to this carrier ID" or something similar

Reply:
Referring to this:
requireNonNull(T obj, String message)

https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html